python/lsst/ap/association/l1dbproto.py

```diff
@@ -0,0 +1,296 @@
+            "l1db-schema.yaml", "L1DBPROTO_DIR")
+        self.l1db_config.column_map = _data_file_name(
+            "l1db-ap-pipe-afw-map.yaml", "AP_ASSOCIATION_DIR")
+        self.l1db_config.extra_schema_file = _data_file_name(
+            "l1db-ap-pipe-schema-extra.yaml", "AP_ASSOCIATION_DIR")
+
+
+class AssociationL1DBProtoTask(pipeBase.Task):
+    """Task wrapping `lsst.l1dbproto` enabling it to be used in ap_association.
+
+    Handles computation of HTM indices, trimming of DIAObject catalogs to the
+    CCD geometry, and assuring input DIASource catalog schemas are compatible
+    with the db.
+    """
+
+    ConfigClass = AssociationL1DBProtoConfig
+    _DefaultName = "association_l1dbroto_task"
+
+    def __init__(self, **kwargs):
+
+        pipeBase.Task.__init__(self, **kwargs)
+        self.indexer = IndexerRegistry[self.config.indexer.name](
+            self.config.indexer.active)
+
+        self._dia_object_afw_schema = make_minimal_dia_object_schema(
+            self.config.filter_names)
+        self._dia_source_afw_schema = make_minimal_dia_source_schema()
+
+        afw_schema = dict(
+            DiaObject=self._dia_object_afw_schema,
+            DiaSource=self._dia_source_afw_schema)
+        self.db = l1db.L1db(config=self.config.l1db_config,
+                            afw_schemas=afw_schema)
+
+    def load_dia_objects(self, exposure):
+        """Load all DIAObjects within the exposure.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            An exposure with a solved WCS representing the area on the sky to
+            load DIAObjects.
+
+        Returns
+        -------
+        dia_objects : `lsst.afw.table.SourceCatalog`
+            Catalog of DIAObjects that are contained with the the bounding
+            box defined by expMd.
+        """
+        bbox = afwGeom.Box2D(exposure.getBBox())
+        wcs = exposure.getWcs()
+        expMd = pipeBase.Struct(
+            bbox=bbox,
+            wcs=wcs,)
+
+        ctr_coord = expMd.wcs.pixelToSky(expMd.bbox.getCenter())
+        max_radius = max(
+            ctr_coord.separation(expMd.wcs.pixelToSky(pp))
+            for pp in expMd.bbox.getCorners())
+
+        indexer_indices, on_boundry = self.indexer.get_pixel_ids(
+            ctr_coord, max_radius)
+        # Index types must be cast to int to work with l1dbproto.
+        index_ranges = [[int(idexer_idx), int(idexer_idx) + 1]
+                        for idexer_idx in indexer_indices]
+        covering_dia_objects = self.db.getDiaObjects(index_ranges)
+
+        output_dia_objects = afwTable.SourceCatalog(
+            covering_dia_objects.getSchema())
+        for cov_dia_object in covering_dia_objects:
+            if self._check_dia_object_position(cov_dia_object, expMd):
+                output_dia_objects.append(cov_dia_object)
+
+        return output_dia_objects.copy(deep=True)
+
+    def _check_dia_object_position(self, dia_object_record, expMd):
```

Comment:
What would be perferable? Should I pass a bbox and wcs or the full exposure? I think the reason I have things the way they are is because I didn't want to have to access the different parts from the exposure over and over while at the same time keeping the wcs and bbox together as one unit.

Reply:
My personal preference would be to pass in a `bbox` and `wcs`.    This function's behavior also surprises me.  If it returned `True`, I would think that it means the object IS definitely in the bbox, not that the bbox/wcs might have been None, and that its not NOT in the bbox.